keps/sig-network/2004-topology-aware-routing/README.md

```diff
@@ -0,0 +1,553 @@
+2. Relatively high in-zone traffic
+
+##### Potential Issues
+1. Weights require API additions
+2. Weights are volatile that increases churn of EndpointSlices (any single
+   endpoint update leads to a weight change for almost every EndpointSlice)
+3. Higher number of EndpointSlices
+4. Lower in-zone traffic than `recommended`
+
+#### Local-Opt
+LO shares the same idea of `recommended` that rebalances endpoints from
+overflowed zones to insufficient zones. Instead of keeping more endpoints than
+expected in a `local EndpointSlice Group`, it shares those overall extra
+endpoints to all zones with an evenly divided weight to achieve better
+overload.
+
+##### Rationale
+1. Relatively high in-zone traffic
+2. Relatively balanced deviation
+
+##### Potential Issues
+1. Weights involved in the tie-breaker stage
+   - Requires API addition
+   - Introduces more volatility
+2. Higher number of EndpointSlices
+3. Lower in-zone traffic than `recommended`
+
+
+### Test Plan
+| Test Name | Test Type | Test Details | Expected Output | Comments |
+| :--- | :--- | :--- | :--- | :--- |
+| Single Approach Test | Unit Test | Small/Large number of nodes | Distribute EndpointSlices as expected | |
+| |  | Balanced/Imbalanced endpoints | Distribute EndpointSlices as expected | |
+| |  | Zero endpoints in some zones | Distribute EndpointSlices as expected | |
+| |  | Invalid annotation | Approach sets to default, distribute EndpointSlices as expected | |
+| `Prefer` Test | Unit Test | Invalid starting threshold | Set starting threshold to default and start `Prefer` approach with condition meets| This is applicable for customized starting threshold only|
+| |  | Endpoints below `starting threshold + padding` | `Prefer` approach will not be conducted | |
+| |  | Endpoints above `starting threshold + padding` | `Prefer` works as expected | |
+| Approach Transition Test | Unit Test | 1.Set annotation to `Prefer` <br>2. Increase endpoints number to above `starting threshold` but below `starting threshold + padding` | `Prefer` will not be conducted | If capable, monitor the churn of EndpointSlices during the rolling update |
+|  |  | 1.Set annotation to `Prefer` <br>2. Increase endpoints number to above `starting threshold + padding` | Distribution will be changed from `Balanced` to `Prefer` | Same as above |
+|  |  | 1.Set annotation to `Prefer` <br>2. Decrease endpoints number to below `starting threshold` but above `starting threshold - padding` | Distribution will remain `Prefer` | Same as above |
+|  |  | 1.Set annotation to `Prefer` <br>2. Decrease endpoints number to below `starting threshold - padding` | Distribution will be changed from `Prefer` to `Balanced` | Same as above |
+|  |  | 1.Set annotation to one of the three <br>2. Set endpoints to above `starting threshold + padding` <br> 3. Manually change the annotation to another approach | Distribution will be changed based on the new value | Step 2 is only needed for `Prefer` |
```

Comment:
We should have a variation with a more specific scenario of Rollback, i.e. 
as a user I have Balanced, I move to Prefer but I'm not happy with the results so I want to move back to Balanced

Reply:
Thanks for pointing it out! Added a new test case with approaches changed from `Balanced` to `Prefer` and switch back to `Balanced`.